src/lib/core/datetime.h

```diff
@@ -113,6 +114,24 @@ datetime_now(struct datetime *now);
 void
 datetime_to_tm(const struct datetime *date, struct tnt_tm *tm);
 
+/**
+ * Parse datetime text in ISO-8601 given format, and construct output
+ * datetime value
+ * @param date output datetime value
+ * @param str input text in relaxed ISO-8601 format (0-terminated)
```

Comment:
Seems like "relaxed ISO-8601" the same as "rfc3339"?
If yes, I suggest mentioning rfc3339, "relaxed ISO-8601" sounds unclear IMO.

Reply:
Not exactly. RFC-3339 is a profile of a strict ISO-8601, not it's extension. We use c-dt, which is ISO8601 conformant in such way that we accept superset of ISO8601 strings, with few reasonable extensions.

Let me clarify a bit:

Years
====

| Basic | Extended | ... | ISO-8601? |
|-------|-----------|-----|-----------|
| `20121224` | `2012-12-24` | Calendar date |  yes |
| `2012359` | `2012-359` | Ordinal date | yes |
| `2012W521` | `2012-W52-1` | Week date | yes |
| `2012Q485` | `2012-Q4-85` | Quarter date |  |

[With extended `c-dt` parser we also support extended years values](https://github.com/tarantool/c-dt/pull/1), which are outside of standard [0001 .. 9999] range
| Tarantool extended values |   |
|------------------------------|---|
| `0000-01-01` | `-0001-12-31` |
| `5879611-07-11` | `-5879610-06-22` |

Time
===

| Basic | Extended |
|-------|-----------|
| `T12` | N/A |
| `T1230` | `T12:30` |
| `T123045` | `T12:30:45` |
| `T123045.123456789` | `T12:30:45.123456789` |
| `T123045,123456789` | `T12:30:45,123456789` |
The time designator `T` may be omitted.

Timezone
======

We do not (yet) support names for standard time-zones (like `MSK`, or `EST`, or `America/New_York` or whatever) we only support numeric offsets at the moment. In a form

| Basic | Extended |
|-------|-----------|
| `Z` | N/A |
| `±hh` | N/A |
| `±hhmm` | `±hh:mm` |
| `GMT±h` |  |
| `GMT±hhmm` |  |
| `GMT±h:mm | `GMT±hh:mm` |
| `UTC±h:mm` | `UTC±hh:mm` |


And relaxed ISO-8601 format is compositions of all relaxed subformats (i.e. date, time, and timezone) putting them together as :

> {relaxed-date} ( `[Tt ]` {relaxed-time} ( `' '` {lenient-time-zone})? )?

But I've got your point - I need to clearly document it somehow (as above) putting it to documentation.